internal/v1/handler_blueprints.go

```diff
@@ -59,11 +66,29 @@ func (h *Handlers) CreateBlueprint(ctx echo.Context) error {
 		return err
 	}
 
+	if !regexp.MustCompile(blueprintNameRegex).MatchString(blueprintRequest.Name) {
+		return ctx.JSON(http.StatusUnprocessableEntity, HTTPErrorList{
+			Errors: []HTTPError{{
+				Title:  "Invalid blueprint name",
+				Detail: "The blueprint name must start and end with an alphanumeric character and may contain alphanumeric characters, underscores, hyphens, and spaces.",
+			}},
+		})
+	}
```

Comment:
You can move the `MustCompile` to the constant (or `init()`) so the regexp is compiled only once, not on every request.